src/client/index.js

```diff
@@ -0,0 +1,51 @@
+import debugFactory from 'debug';
+import getData from './get-data';
+import requireData from './require-data';
+
+const debug = debugFactory( 'fresh-data:api-client' );
+
+export default class ApiClient {
+	constructor( api, key ) {
+		this.api = api;
+		this.key = key;
+		this.requirementsByComponent = new Map();
+		this.requirementsByEndpoint = {};
+		this.methods = mapMethods( api.methods, key );
+		this.setState( {} );
+		debug( 'New ApiClient "' + key + '" for api: ', api );
+	}
+
+	setState = ( state ) => {
+		if ( this.state !== state ) {
+			this.state = state;
+			// TODO: Check and update endpoint requirements.
+		}
+	}
+
+	getData = ( endpointPath, params ) => {
```

Comment:
The double `getData`'s here had me scratching my head for a second. Should this one be called `getAPIData` maybe or something else?